Marlin/src/HAL/HAL_AVR/fast_pwm.cpp

```diff
@@ -23,7 +23,7 @@
 
 #include "../../inc/MarlinConfigPre.h"
 
-#if ENABLED(FAST_PWM_FAN)
+#if ANY(FAST_PWM_FAN, SPINDLE_LASER_PWM) //FAST_PWM_FAN Shouldn't be dependency for pwm control; but sanguino_atmega1284p won't build this file; means hal is borkened
```

Comment:
`FAST_PWM_FAN` is not a dependency. Rather, it is the only feature that requires this file. So when it is not enabled, the file is not compiled. This is common in Marlin because during the overhaul of the source code we wanted to make it obvious what features require what code so that we could work to reduce dependencies and better organize the codebase.

Reply:
Should there be a `NEED_PWM` metadefine generated in much the same way `HAS_CUTTER`?

Then new hard-pwm in hal just needs to check for `NEED_PWM`

Reply:
If these `EITHER(FAST_PWM_FAN, SPINDLE_LASER_PWM)` lines multiply to more than a few, then we can add a conditional.